### PR TITLE
Fix Azure Remote Rendering tests by adding now missing dependencies and changed conversion error messages

### PR DESCRIPTION
**Pull request by MichaelZp0**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by MichaelZp0** [comment]
So, that change isn't great. The test fixes are absolutely fine, but the test framework fixes are kinda bad.

Problem is, that the new HTTP client azure-core-http-jdk-httpclient doesn't work with Java 1.8. From the matrix config file it looks like it shouldn't be run in tandem, only with 1.17, but it somehow does anyway. So I disabled Java 1.8 tests for now and will contact the person that added the azure-core-http-jdk-httpclient tests to see if we can find a better solution.

I would still go forward with this PR, as we are currently testing nothing and there is a risk of missing breaks in the releases for normal ARR (not the java SDK itself).

**Review by ChristopherManthei — APPROVED** [review]

File: sdk/remoterendering/azure-mixedreality-remoterendering/pom.xml [hunk 1]
```diff
@@ -52,6 +52,16 @@
       <version>1.11.19</version> <!-- {x-version-update;com.azure:azure-core-http-okhttp;dependency} -->
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-http-jdk-httpclient</artifactId>
+      <version>1.0.0-beta.11</version> <!-- {x-version-update;com.azure:azure-core-http-jdk-httpclient;dependency} -->
+    </dependency>
```
Comment (alzimmermsft) [inline sdk/remoterendering/azure-mixedreality-remoterendering/pom.xml:59]: To have the tests work properly with Java 8, you'll want to remove this dependency and leave what is added later on with the JDK 12+ profile

Comment (mssfang) [reply sdk/remoterendering/azure-mixedreality-remoterendering/pom.xml:59]: <profiles>
    <profile>
      <id>java12plus</id>
      <activation>
        <jdk>[12,)</jdk>
      </activation>
      <dependencies>
        <dependency>
          <groupId>com.azure</groupId>
          <artifactId>azure-core-http-jdk-httpclient</artifactId>
          <version>1.0.0-beta.11</version> 
          <scope>test</scope>
        </dependency>
      </dependencies>
    </profile>
  </profiles>

File: sdk/remoterendering/azure-mixedreality-remoterendering/pom.xml [hunk 2]
```diff
@@ -52,6 +52,11 @@
       <version>1.11.19</version> <!-- {x-version-update;com.azure:azure-core-http-okhttp;dependency} -->
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-http-vertx</artifactId>
+      <version>1.0.0-beta.16</version> <!-- {x-version-update;com.azure:azure-core-http-vertx;dependency} -->
```
Comment (alzimmermsft) [inline sdk/remoterendering/azure-mixedreality-remoterendering/pom.xml:58]: This is missing test scope

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**